lib/tessel.js

```diff
@@ -441,6 +442,17 @@ function Tessel(options) {
     return new Pin(Object.assign({}, config, ToPortIdentity(index)));
   });
 
+  // REPL control:
+  // - If we detect that we're running on-board the Tessel, disable the repo
+  // - If the user specifies a repl option, use it no matter what
+  var tesselPathRegex = /^\/app\/remote-script/;
+  if (tesselPathRegex.test(__dirname)) {
+    this.repl = false;
+  }
+  if (typeof options.repl !== "undefined") {
+    this.repl = options.repl;
+  }
```

Comment:
It wasn't obvious, but I did notice it. Should we assume that the `repl` stays `true` under `t2 run` and only forcibly disable under `t2 push` ?